features/cellular/easy_cellular/CellularConnectionFSM.cpp

```diff
@@ -426,15 +435,26 @@ void CellularConnectionFSM::state_registering()
 {
     _cellularDevice->set_timeout(TIMEOUT_NETWORK);
     if (is_registered()) {
-        // we are already registered, go to attach
-        enter_to_state(STATE_ATTACHING_NETWORK);
-    } else {
-        bool auto_reg = false;
-        nsapi_error_t err = is_automatic_registering(auto_reg);
-        if (err == NSAPI_ERROR_OK && !auto_reg) { // when we support plmn add this :  || plmn
-            // automatic registering is not on, set registration and retry
+        if (_plmn && _retry_count == 0) {
```

Comment:
Any chance to know current PLMN? It's possible that we might be registered in correct PLMN already.